src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs

```diff
@@ -54,37 +54,48 @@ public static HostType GetHostType()
             return hostType;
         }
 
-        public static async Task CallWithRequestTracing(bool tracingEnabled, RequestType requestType, HostType hostType, Action clientCall)
+        public static bool IsDevEnvironment()
         {
-            await CallWithRequestTracing(tracingEnabled, requestType, hostType, () =>
+            try
+            {
+                string envType = Environment.GetEnvironmentVariable(RequestTracingConstants.AspNetCoreEnvironmentVariable) ??
+                                    Environment.GetEnvironmentVariable(RequestTracingConstants.DotNetCoreEnvironmentVariable);
+                if (envType != null && envType.Equals(RequestTracingConstants.DevelopmentEnvironmentName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+            catch (SecurityException) { }
+
+            return false;
+        }
+
+        public static async Task CallWithRequestTracing(bool tracingEnabled, RequestType requestType, RequestTracingOptions requestTracingOptions, Action clientCall)
+        {
+            await CallWithRequestTracing(tracingEnabled, requestType, requestTracingOptions, () =>
             {
                 clientCall();
                 return Task.CompletedTask;
             }).ConfigureAwait(false);
         }
 
-        public static async Task CallWithRequestTracing(bool tracingEnabled, RequestType requestType, HostType hostType, Func<Task> clientCall)
+        public static async Task CallWithRequestTracing(bool tracingEnabled, RequestType requestType, RequestTracingOptions requestTracingOptions, Func<Task> clientCall)
```

Comment:
What if RequestTracingOptions is null.

Reply:
If request tracing is enabled, we create a new instance of `RequestTracingOptions` in `AzureAppConfigurationProvider` constructor. So `requestTracingOptions` cannot be null.

If request tracing is disabled, we won't hit the code path that is using the `requestTracingOptions` anyway.
But I can still add another null check to be explicit.